core/src/main/java/edu/wpi/grip/core/Connection.java

```diff
@@ -112,6 +114,27 @@ public void remove() {
     eventBus.post(new ConnectionRemovedEvent(this));
   }
 
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("src", simpleString(outputSocket))
+        .add("dst", simpleString(inputSocket))
+        .toString();
+  }
+
+  private static String simpleString(Socket<?> socket) {
```

Comment:
Why is this method in the connection class instead of a helper method on the `Socket` object??
Throw this in a default method on the `Socket` interface.